assets/js/translator.js

```diff
@@ -1252,6 +1249,52 @@ function autoSelectDstLang() {
     $('#dstLangSelect').val(curDstLang).change();
 }
 
+function setDefaultSrcLang() {
+    // Default for new users is first available Browser Preference Language pair
+    var browserLangs = window.navigator.languages; // Chrome, Mozilla and Safari
+    var ieLang = window.navigator.userlanguage || window.navigator.browserlanguage; // Internet Explorer
+    var prefSrcLang;
+    var i;
+    for(i = 0; i < browserLangs.length; ++i) {
+        var browserLang = getLangCode(browserLangs[i]);
+        if(checkLangPairAvailable(browserLang)) {
+            prefSrcLang = browserLang;
+            break;
+        }
+    }
+    if(!prefSrcLang) {
+        if(ieLang) {
+            var iePrefLang = getLangCode(ieLang);
+            if(checkLangPairAvailable(iePrefLang)) {
+                prefSrcLang = iePrefLang;
+            }
+        }
+    }
+    if(!prefSrcLang) {
+        noPrefSrcLang();
+    }
+    curSrcLang = iso639CodesInverse[prefSrcLang];
+    handleNewCurrentLang(curSrcLang, recentSrcLangs, 'srcLang');
+    autoSelectDstLang();
+
+    // Check if the Language pair Exist or Not
+    function checkLangPairAvailable(lang) {
+        return languages[lang] && pairs[iso639CodesInverse[lang]];
+    }
+
+    // Set curSrcLang as locale or first available pair, if locale not available
+    function noPrefSrcLang() {
```

Comment:
Please inline this function. It's only used once and is relatively small.

Reply:
Yeahh its it should be. :)